### PR TITLE
Some refactoring of MUMS for speed and readability

### PR DESCRIPTION
**Pull request by akmiller01**
Saves about 6 seconds on a full data download, and no longer requires any arguments (unless you want a faster download with fewer donors/years):

```
> start = Sys.time()
> dat = MUMS()
> end = Sys.time()
> message(end - start)
15.353663444519
```

As compared to original:

```
> start = Sys.time()
> dat = MUMS(donors="", Start="2011", End="2022")
> end = Sys.time()
> message(end - start)
21.3280854225159
```

Otherwise, this is difficult to improve. Really nice code!

### CODE REVIEW

File: MUMS.R [hunk 1]
```diff
@@ -60,10 +62,6 @@ MUMS = function(donors,Start,End,core=T,prices="current",disb=T,channel=""){
 	dat = Map(namefunk, dat, 1:length(dat))
 	dat = as.data.frame(rbindlist(dat)) %>% pivot_wider(names_from=Var,values_from=value)
 	dat = remove_constant(dat)
-	names(dat)[names(dat)=="TIME_PERIOD"] = "year"
+	setnames(dat, "TIME_PERIOD", "year")
```
Comment (euanritchieDI) [inline MUMS.R:67]: ahh I didn't know this existed, I have loads of scripts where names(df)[names(df)=="X"]= "Y" take up like 10+ lines, and always thought there must be a better way. 
Just checking though is this a change for readability or performance?

**Review by euanritchieDI — CHANGES_REQUESTED** [review]
Thanks for this. Don't really care about changes, but requesting them to see what happens. 
It looks like I can only bulk accept or not, rather than (for example) accepting the package installation but not the attrfunk changes?

File: MUMS.R (hunk 1 above)
Comment (akmiller01) [reply MUMS.R:67]: It's a bit of both. I don't know how it affects how long it takes to execute, but I like how verbose `setnames` is in terms of a function, and the function also first checks that the names you're trying to change exist and will throw an error if they don't.

You can also pass arrays to the second and third arguments, so you can rename a bunch of variables at once.

**Comment by akmiller01** [comment]
> Thanks for this. Don't really care about changes, but requesting them to see what happens. It looks like I can only bulk accept or not, rather than (for example) accepting the package installation but not the attrfunk changes?

I think in the settings for this PR I gave you permission to modify the contents. But you would be modifying them in your IDE after checking the correct branch/ref out and committing them back here. It's also only an option because the PR comes from my fork instead of a branch within your repo, otherwise GitHub presumes you'd already have the ability to modify the branch.

What I tend to do when I want a PR author to change something is go into the files changed and start a full review, and then you can comment on each individual line you want changed.

**Pull request merged**